spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -341,37 +361,51 @@ public void run() {
 				else {
 					for (Message<T> message : messages) {
 						do {
-							try {
-								if (this.listener instanceof PulsarAcknowledgingMessageListener) {
-									this.listener.received(this.consumer, message, this.ackMode.equals(AckMode.MANUAL)
-											? new ConsumerAcknowledgment(this.consumer, message) : null);
-								}
-								else if (this.listener != null) {
-									this.listener.received(this.consumer, message);
-								}
-								if (this.ackMode.equals(AckMode.RECORD)) {
-									handleAck(message);
-								}
-								inRetryMode.compareAndSet(true, false);
+							Observation observation;
+							if (!this.containerProperties.isObservationEnabled() || this.observationRegistry == null) {
+								observation = Observation.NOOP;
 							}
-							catch (Exception e) {
-								if (this.pulsarConsumerErrorHandler != null) {
-									invokeRecordListenerErrorHandler(inRetryMode, message, e);
-								}
-								else {
+							else {
+								observation = PulsarListenerObservation.LISTENER_OBSERVATION.observation(
+										this.containerProperties.getObservationConvention(),
+										DefaultPulsarListenerObservationConvention.INSTANCE,
+										new PulsarMessageReceiverContext(message, getBeanName()),
```

Comment:
Same have about `createNotStarted()` with a `Supplier` instead of this `if..else`

Reply:
Same as https://github.com/spring-projects-experimental/spring-pulsar/pull/146#discussion_r981645315

Reply:
Resolving and will address (or not) based on comment above.